### PR TITLE
Fix for mmap of empty file

### PR DESCRIPTION
**Pull request by sbc100**
Closes: #22202

### CODE REVIEW

File: test/other/test_mmap_empty.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  char *mapped = (char*) mmap(NULL, 0, PROT_READ, MAP_PRIVATE, fd, 0);
+  assert(mapped == MAP_FAILED);
+  printf("mmap failed with: %s\n", strerror(errno));
+  assert(errno == EINVAL);
+
+  // same if we ask for PAGE_SIZE bytes
+  mapped = (char*) mmap(NULL, PAGE_SIZE, PROT_READ, MAP_PRIVATE, fd, 0);
+  assert(mapped == MAP_FAILED);
+  printf("mmap failed with: %s\n", strerror(errno));
+  assert(errno == EINVAL);
```
Comment (kripken) [inline test/other/test_mmap_empty.c:31]: Should we also check the flip case, of the file *not* being empty, but we ask for 0 bytes?

Comment (sbc100) [reply test/other/test_mmap_empty.c:50]: Done.

Also tested and fixed under wasmfs

File: test/other/test_mmap_empty.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+int main() {
+  struct stat st;
+
+  create_file("empty_file", "");
+  create_file("nonempty_file", "some content");
+
+  int fd = open("empty_file", O_RDONLY);
+  int fd2 = open("nonempty_file", O_RDONLY);
+
+  // "fd" should be an empty file, "fd2" non-empty;
```
Comment (kripken) [inline test/other/test_mmap_empty.c:30]:
```suggestion
  // "fd" should be an empty file, "fd2" non-empty.
```

File: test/other/test_mmap_empty.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  assert(errno == EINVAL);
+
+  mapped = (char*) mmap(NULL, 0, PROT_READ, MAP_PRIVATE, fd2, 0);
+  printf("mmap -> %p\n", mapped);
+  assert(mapped == MAP_FAILED);
+  printf("mmap failed with: %s\n", strerror(errno));
+  assert(errno == EINVAL);
+
+  // mapping a non-zero number of pages should succeed, even for empty files.
+  // On native system this result in pages beyond the length of the file
```
Comment (kripken) [inline test/other/test_mmap_empty.c:53]:
```suggestion
  // On native systems this results in pages beyond the length of the file
```

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
(disabled auto-merge because I had comments, but feel free to ignore them if you disagree)

**Pull request merged**